contracts/TokenFactory.sol

```diff
@@ -304,10 +306,17 @@ contract TokenFactory is Initializable, AccessControlUpgradeable, UUPSUpgradeabl
         emit BurnTokenAndMintWinner(msg.sender, tokenAddress, winnerToken, burnedAmount, mintAmount, fee, block.timestamp);
     }
 
-    function withdrawFee() external onlyRole(DEFAULT_ADMIN_ROLE) {
-        uint256 feeWithdrawable = feeAccumulated - feeWithdrawn > address(this).balance ? address(this).balance : feeAccumulated - feeWithdrawn;
-        (bool success, ) = payable(msg.sender).call{value: feeWithdrawable}("");
-        require(success, "transfer failed");
+    function withdrawFee(address recipient) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        uint256 availableFee = feeAccumulated > feeWithdrawn ? feeAccumulated - feeWithdrawn : 0;
```

Comment:
fee accumulated should not be less than fee withdrawn. If it happens, some implementation is wrong. Use assert here?